MetaMask/metamask-mobile
# fix: token decimals fetched from the chain

## **Description**
The problem was that the user was able to insert wrong the token decimals values, and with that would get the wrong tokens amounts in their wallet.

* Added a new state to lock the text inputs of symbol and decimals when the symbol
* Decimals and names are fetchable from the chain
* Reset all the states when the address text input changed when an address does not have 42 chars
* Added styling to address the value of symbols and decimals text input when fetched from the chain

It was also added to the wallet_watchAsset JSON rpc method
* overrides token decimals and symbol if they aare fetched from the chain


## *Note for review*
*  On RPCMethodMiddleware  file `wallet_watchAsset` method
 * On the 207 line validateCustomTokenAddress of AddCustomToken  

## **Manual testing steps**
Scenario 1:
_1. Step1: Go to import token on wallet view
_2. Step2: change tab to custom token
_3. Step3: add `0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599` (WBTC) as token address
_4. Step4: It should block all the symbol and decimals field 


Scenario 2
 Step1: Go to in app browser
 Step2: Go to coingecko website
 Step3: Browse WBTC
 Step4: look for the MM icon 
 Step5: Add it via coin gecko
 Step6: It should work as expected

## **Screenshots/Recordings**
* importing token on custom token screen
http://recordit.co/ENT3D6Y0wn

* import token on coin gecko
https://recordit.co/Ikwc5adYZn

* import token on test dapp with wrong symbol and decimals (WBTC contract)
https://recordit.co/zoixJsK4Sc

* import WBTC on polygon network
https://recordit.co/zTtxIthF4C

## **E2E build**
https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/1d5eb4d0-faff-456b-b66b-6e55718f4cdf


## **Related issues**

_Fixes #

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
